xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/macros.vm

```diff
@@ -768,16 +768,21 @@ $fn
 #macro(panelheader $title)
 ## Note: We pass the Panel name as an HTML class attribute so that it's possible to style
 ## the Panels selectively using CSS.
-#set ($specialClassAttribute = '')
-#if ($paneldoc)
-  #set ($specialClassAttribute = $util.convertToAlphaNumeric($paneldoc.name))
+#set ($defaultDocumentName = $services.model.getEntityReference('DOCUMENT', 'default').name)
+#if ($paneldoc.documentReference.name == $defaultDocumentName)
+  #set ($serializedDocument = $!services.model.serialize($paneldoc.documentReference.parent, 'compactwiki'))
+#else
+  #set ($serializedDocument = $!services.model.serialize($paneldoc.documentReference, 'compactwiki'))
 #end
+#set ($specialClassAttribute = $stringtool.toAlphaNumeric("$!serializedDocument"))
```

Comment:
Maybe explain that CSS class name can only contain alphanumeric characters and provide and example, e.g. "xwiki:A.B.WebHome" will generate "AB".